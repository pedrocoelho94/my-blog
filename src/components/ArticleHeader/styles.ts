import styled, { css } from 'styled-components'
import * as HeadingStyles from 'components/Heading/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles.Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.font.sizes.large};
    }
  `}
`

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: calc(${theme.font.sizes.medium} * 0.9);
  `}
`

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`
