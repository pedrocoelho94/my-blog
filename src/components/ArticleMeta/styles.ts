import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child):after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.2s ease-in-out;

      &:hover {
        filter: brightness(0.5);
      }
    }
  `}
`
