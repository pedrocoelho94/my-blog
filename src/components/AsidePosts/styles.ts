import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Content = styled.a`
  ${({ theme }) => css`
    display: flex;
    gap: 1rem;
    height: 100%;
    cursor: pointer;
    text-decoration: none;
    margin: ${theme.spacings.medium} 0;
  `}
`

export const Cover = styled.img`
  border-radius: 0.8rem;
  width: 10rem;
  height: 10rem;
  object-fit: cover;
`

export const Title = styled.h4`
  ${({ theme }) => css`
    font-size: 1.5rem;
    height: 100%;
    margin: 0;
    color: ${theme.colors.darker};

    &:hover {
      text-decoration: underline;
    }
  `}
`
