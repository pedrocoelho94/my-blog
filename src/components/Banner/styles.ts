import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  position: relative;
`

export const Content = styled.div`
  filter: brightness(0.95);
  transition: filter 0.2s ease-in-out;

  &:hover {
    filter: brightness(1.2);
  }
`

type imgProps = {
  src: string
}

export const Image = styled.div<imgProps>`
  ${({ src }) => css`
    width: 100%;
    height: 30rem;
    background-image: url(${src});
    background-position: center center;
    background-size: cover;

    ${media.greaterThan('small')`
      height: 40rem;
    `}

    ${media.greaterThan('medium')`
      height: 45rem;
    `}
  `}
`
export const Caption = styled.div`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.medium};
    background-color: rgba(0, 0, 0, 0.4);
    position: absolute;
    bottom: 0;
    left: 0;
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    margin-top: 0;
    color: ${theme.colors.white};

    ${media.lessThan('small')`
      font-size: 2rem;
      margin-bottom: 0;
    `}
  `}
`

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    font-weight: inherit;
    margin-bottom: 0;
    color: ${theme.colors.white};

    ${media.lessThan('small')`
      display: none
    `}
  `}
`
