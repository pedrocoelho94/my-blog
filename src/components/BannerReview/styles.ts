import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Poster = styled.div`
  position: relative;
  width: 100%;
  max-width: 25rem;
  height: 100%;
  max-height: 35rem;
  transition: all 0.2s ease-in-out;
  overflow: hidden;
  border-radius: 5px;
`

export const Image = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 5px;
  transition: all 0.2s ease-in-out;
  transform: scale(1.05);

  &:hover {
    transform: scale(1.1);
  }
`

type RatingProps = {
  rating: number
}

const RatingModifiers = {
  good: () => `
    background-color: #01d200;
  `,
  ok: () => `
    background-color: #ffb300;
  `,
  bad: () => `
    background-color: #ff7e00;
  `,
  terrible: () => `
    background-color: #df0700;
  `
}

export const Rating = styled.div<RatingProps>`
  ${({ theme, rating }) => css`
    position: absolute;
    bottom: 0;
    right: 0;
    z-index: 5;
    color: ${theme.colors.white};
    font-size: 1.8rem;
    font-weight: bold;
    width: 3.5rem;
    height: 3.5rem;
    display: flex;
    border-radius: 5px;

    span {
      width: 100%;
      text-align: center;
      justify-self: center;
      align-self: center;
    }

    ${rating >= 7.5 && RatingModifiers.good()}
    ${rating >= 5 && rating < 7.5 && RatingModifiers.ok()}
    ${rating >= 2.5 && rating < 5 && RatingModifiers.bad()}
    ${rating < 2.5 && RatingModifiers.terrible()}
  `}
`
