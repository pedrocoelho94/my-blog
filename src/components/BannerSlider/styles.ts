import styled, { css } from 'styled-components'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    .slick-dots {
      list-style: none;
      display: flex !important;
      align-items: center;
      justify-content: center;
      margin-top: ${theme.spacings.small};
    }

    li {
      background: ${theme.colors.primary};
      width: 1.2rem;
      height: 1.2rem;
      border-radius: 100%;
      display: flex;
      align-items: center;
      justify-content: center;
      margin: 0 ${theme.spacings.xsmall};
      cursor: pointer;

      &.slick-active {
        background: ${theme.colors.secondary};
      }

      button {
        opacity: 0;
        width: 1.2rem;
        height: 1.2rem;
        cursor: pointer;
      }
    }
  `}
`
