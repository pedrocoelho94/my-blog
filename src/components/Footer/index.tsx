import TextComponent from 'components/TextComponent'
import * as S from './styles'

export type FooterProps = {
  footerHtml: string
}

const Footer = ({ footerHtml }: FooterProps) => (
  <S.Footer>
    <TextComponent>{footerHtml}</TextComponent>
  </S.Footer>
)

export default Footer
