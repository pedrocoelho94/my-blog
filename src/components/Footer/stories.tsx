import { Story, Meta } from '@storybook/react'
import Footer, { FooterProps } from '.'

export default {
  title: 'Footer',
  component: Footer,
  args: {
    footerHtml: `<p><a href="https://github.com/pedrocoelho94">Feito com ‚ù§ por Pedro Coelho</a></p>`
  }
} as Meta

export const Default: Story<FooterProps> = (args) => (
  <div style={{ marginTop: '200px' }}>
    <Footer {...args} />
  </div>
)
