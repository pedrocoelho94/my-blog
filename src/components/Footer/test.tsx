import { screen } from '@testing-library/react'
import { renderWithTheme } from 'styles/renderWithTheme'

import Footer from '.'

describe('<Footer />', () => {
  it('should render the footer', () => {
    const { container } = renderWithTheme(
      <Footer footerHtml="<h1>Hello</h1>" />
    )

    expect(screen.getByRole('heading', { name: 'Hello' })).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
