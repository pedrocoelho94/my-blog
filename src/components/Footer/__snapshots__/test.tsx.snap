// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the footer 1`] = `
.c1 {
  font-size: calc(1.6rem + 0.2rem);
  line-height: 1.5;
}

.c1 p {
  margin: 4.0rem 0;
}

.c1 a,
.c1 a:visited,
.c1 a:link {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c1 a:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

.c1 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c1 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FAFAFA;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c1 pre code {
  color: inherit;
  background: inherit;
}

.c1 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c1 img {
  max-width: 100%;
}

.c1 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c1 .image figcaption {
  font-size: 1.6rem;
  padding: 0.8rem;
  text-align: center;
  line-height: 1.3;
}

.c1 ul,
.c1 ol {
  margin: 2.4rem 4.0rem;
}

.c1 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c1 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c1 table td,
.c1 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c1 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c0 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
}

.c0 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width:768px) {
  .c1 .image-style-side {
    float: right;
    max-width: 50%;
    margin: 2.4rem;
  }
}

@media (max-width:768px) {
  .c1 {
    font-size: 2rem;
  }
}

<div>
  <footer
    class="c0"
  >
    <div
      class="c1"
    >
      <h1>
        Hello
      </h1>
    </div>
  </footer>
</div>
`
